pulp_rpm/app/downloaders.py

```diff
@@ -94,3 +104,88 @@ async def _run(self, extra_data=None):
         if self._close_session_on_finalize:
             self.session.close()
         return to_return
+
+
+class UlnDownloader(RpmDownloader):
+    """
+    Custom Downloader for ULN repositories.
+
+    Args:
+        uln_user (str): Username for authentification in ULN network
+        uln_pwd (str): password for authentification in ULN network
+        server_url (str): ULN server url. If no url is given, use default oracle
+            ULN server url `https://linux-update.oracle.com/`
+
+    Raises:
+        NoUlnCredentials: If no or not valid ULN credentials are given, this Error will be displayed
+    """
+
+    def __init__(self, *args, uln_user=None, uln_pwd=None, server_url=None, **kwargs):
+        """
+        Initialize the downloader for ULN repositories.
+
+        If no server URL is given, use the server url from oracle linux.
+        """
+        self.uln_user = uln_user
+        self.uln_pwd = uln_pwd
+        if server_url:
+            self.server_url = server_url
+        else:  # if no server_url is given, use oracle linux url
+            self.server_url = "https://linux-update.oracle.com/"
```

Comment:
Would it make sense to create a separate type of remote for ULN? (Something like `UlnRemote` or `UlnRpmRemote`).
Then we could make the `server_url` field (or whatever we will call it) a required field, and would not require such a fallback. Alternatives might be to raise an exception when users supply a ULN URL without supplying a server URL, or to configure the server URL somewhere other than the remote.

Reply:
This has been answered in the affirmative.